### DISCUSSION TITLE
Pull through the changed labels for fields on the entries table

### DISCUSSION THREAD

**Discussion by dgtlcrftrs** [craftcms/cms#8903] [Ideas] [answered]
### Description

For each 'section' you can choose what table columns show for the list of entries. There should be a way to either change these from their default or show the label set on the field itself when overriding it.

For example, I have a field called 'Secondary Copy' which I'm using in multiple places in different ways. I have changed the label for each use case. i.e on a quotes entry it's 'company name'. On the table when viewing the quote entries it still says 'Secondary Copy' with no way of overriding this.

**Comment by brandonkelly** [comment]
This would need to be done as an additional setting within the Customize Sources modal on the element index (similar to the “Heading Column Label” setting added in 3.5), as a single field could have multiple overridden names within the same source (especially under “All entries”, but even within a single section as well).

**Reply by Anubarak** [reply]
Opening this issue again.
Would it be an option for you to at least include a setting in the field layout in order to use the `$element->label()` instead of `$field->name` here?
https://github.com/craftcms/cms/blob/4.5/src/services/ElementSources.php#L336-L338

```PHP
$label = $fieldLayout->useAlteredLabel()? $element->label() : $field->name;
$event->attributes["field:$field->uid"] = [
    'label' => Craft::t('site', $label),
];
```


I know that would lead to error in some cases and it would not be a perfect solution, but the current behavior is mostly always wrong for me. If there was such a setting, the field labels would not always be wrong.

Currently I always have to use `DefineSourceTableAttributesEvent` and basically copy your entire code from this function just to replace `'label' => Craft::t('site', $field->name),` with `'label' => Craft::t('site', $element->label()),`

**Reply by brandonkelly** [reply]
Came up with a compromise: https://github.com/craftcms/cms/discussions/8903#discussioncomment-5892928

**Comment by Anubarak** [comment]
@brandonkelly I guess it would be helpful to at least grab the labels for the first entry type when filtered via sections/category groups/sources in general. 

I did the same with my "Relabel" plugin and searched for different labels in all entry types and replaced those accordingly. 
If there are several different one I just used the first I found. While this is of course not totally correct it's at least better than showing the original one. 

I mostly create a set of generic fields with the name "Lightswitch" or so in order to find those really fast when creating a new field layout 1-2 years after the project was created. If I would currently uninstall my (now mostly useless) Relabel plugin my element indexes would all be confusing.
My plugins one and only single purpose now is to replace field names in the element index, the sources modal and the menu btn for sorting 😅

**Comment by brandonkelly** [comment, marked as answer]
I’ve just made a change for Craft 4.5: field label overrides will be used when all of the field layouts for the selected source are set to the same label. (1890d972ff55493bda67a94841d6a9ad48adfa80)

So if you’re viewing a section that only has one entry type, then its label overrides will definitely be used. If a section has two field layouts, and a field is used across both of them, its label override will only be used if both layouts have the same override value.

**Reply by brandonkelly** [reply]
Craft 4.5.0 is out with that change.
